### ISSUE TITLE
Feature Request: file sync (upload/download/delete) notifications

### ISSUE THREAD

**Issue by rubiksdot** [abraunegg/onedrive#2760]
### Is your feature request related to a problem? Please describe.

Currently it is not possible to see if a sync (file upload/download/delete) has happened when onedrive is running as a service in the background.

### Describe the solution you'd like

I would like to see if files got uploaded/downloaded or deleted (on either side). This would give some comfort in knowing, for eg, when to turn off the laptop or to tell someone to look on onedrive as the file should be there.

If the file is big enough it'd be nice to get a start/finish notification.

### Describe alternatives you've considered

Manually checking but, well, that's a pain in the butt. I should be able to get my computer to indicate what's happening.

### Additional context

Currently running 2.4.25 on Debian Bookworm.

**Comment by abraunegg** [comment]
@rubiksdot 
GUI events and notifications for these events are already available.

Have you enabled them ?

**Comment by rubiksdot** [comment]
Oooh. My apologies. How? I checked the man page but I didn't spot anything.

**Comment by abraunegg** [comment]
Please read *all* documentation - not just the man page

**Comment by rubiksdot** [comment]
Ok. I tried. I thought it was `min_notify_changes` but setting that to 1 did nothing and I tried both syncing to one drive and from it.

I looked at the code and maybe this is the wrong place but

```
if (!downloadFailed) {
			writeln("done.");
			log.fileOnly("Downloading file ", path, " ... done.");
		} else {
			writeln("failed!");
			log.fileOnly("Downloading file ", path, " ... failed!");
		}
```

does not appear to indicate that the functionality is there.

I do get notifications of onedrive starting up and all but that's about it.

Honestly, am I being a bit thick/blind here or is the functionality not there and we're talking past each other?

**Comment by abraunegg** [comment]
@rubiksdot 
So we are talking a number of things:
* claim that it is not possible to see when file actions are done
  * This is possible via enabling logging and parsing the log file
* notifying via the GUI of those actions
  * the code in v2.4.x lacks this - correct, however my focus has been v2.5.x over the last 18 months - sorry for the misdirect

**Comment by rubiksdot** [comment]
All good. Glad it got cleared up in the end. :) Also, understand that v2.5 would take priority. :)

**Comment by mstoetzer** [comment]
Maybe a bit related: is there a way to get the sync status in one line?
Something like `onedrive --display-sync-status` that returns "okay" or such.
I'm thinking about adding this the command line in the shell prompt.

**Comment by abraunegg** [comment]
This is already in the client

**Comment by abraunegg** [comment]
@rubiksdot 
This is now implemented in v2.5.0 RC3.

Please review: https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-3/docs/usage.md#gui-notifications

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
